### PR TITLE
treewide: update nRF9161DK external flash.

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Update nRF9161DK in following samples and applications:
* asset_tracker_v2
* serial_lte_modem
* lwm2m_client
* modem_shell
* modem_trace_flash
* nrf_cloud_multi_servic
* nrf_cloud_rest_fota

Might have to revert the changes to ATv2 with the external flash DTS for twister to work.

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]

**Review by maxd-nordic — APPROVED** [review]

**Review by tomi-font — APPROVED** [review]

**Review by jheiskan81 — APPROVED** [review]

File: applications/asset_tracker_v2/boards/nrf9160dk_nrf9160_ns_0_14_0.overlay [hunk 1]
```diff
@@ -9,6 +9,15 @@
...
 	};
+
+	aliases {
+		ext-flash = &mx25r64;
+	};
+};
+
+&spi3 {
+	cs-gpios = <&arduino_header 16 GPIO_ACTIVE_LOW>, /* D10 */
+		   <&gpio0 25 GPIO_ACTIVE_LOW>;
```
Comment (gregersrygg) [inline applications/asset_tracker_v2/boards/nrf9160dk_nrf9160_ns_0_14_0.overlay:20]: This is the default for 0.14.0. Is it necessary to re-define here?

File: applications/asset_tracker_v2/boards/nrf9161dk_nrf9161_ns.overlay [hunk 2]
```diff
@@ -42,12 +48,23 @@
...
 	};
 };
 
-&gd25lb256 {
+&spi3 {
+	compatible = "nordic,nrf-spim";
+	status = "okay";
+	cs-gpios = <&arduino_header 16 GPIO_ACTIVE_LOW>, /* D10 */
+		   <&gpio0 20 GPIO_ACTIVE_LOW>;
+};
```
Comment (gregersrygg) [inline applications/asset_tracker_v2/boards/nrf9161dk_nrf9161_ns.overlay:66]: This is the same as the default. Why redefine?

File: samples/cellular/lwm2m_client/boards/nrf9161dk_nrf9161_ns_0_7_0.overlay [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+/ {
+	chosen {
+		nordic,pm-ext-flash = &gd25lb256;
+	};
+};
+
+&gd25lb256 {
+	status = "okay";
+
```
Comment (gregersrygg) [inline samples/cellular/lwm2m_client/boards/nrf9161dk_nrf9161_ns_0_7_0.overlay:17]: Extra newline

File: samples/cellular/modem_trace_flash/boards/nrf9161dk_nrf9161_ns_0_7_0.overlay [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+&gd25lb256 {
+	status = "okay";
+};
+
+/* Enable high drive mode for the SPI3 pins to get a square signal at 8 MHz */
+&spi3_default {
+	group1 {
+		nordic,drive-mode = <NRF_DRIVE_H0H1>;
+	};
+};
```
Comment (gregersrygg) [inline samples/cellular/modem_trace_flash/boards/nrf9161dk_nrf9161_ns_0_7_0.overlay:28]: This is already enabled by default in `nrf9161dk_nrf9161_common-pinctrl.dtsi`.

**Review by gregersrygg — APPROVED** [review]
LGTM. Just some nits.

**Review by MarkusLassila — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Comment by de-nordic** [comment]
West got outdated.

**Review by glarsennordic — APPROVED** [review]

File: west.yml [hunk 5]
```diff
@@ -61,7 +61,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 9f513ec4ec1520912f65e7d7d938e373b5e04bcd
+      revision: pull/1329/head
```
Comment (rlubos) [inline west.yml:64]: `8c7463b5230ce09b74d75e1b963d0c262d92fa73`

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
CI passed on rerun

**Comment by jayteemo** [comment]
@eivindj-nordic my 9161 DK (0.7.0) appears to have the `gd25lb256` flash device, as its ID is `c8 67 19`.

were older versions populated with the `lb` flash device?
i see the schematic references `wb`...  which DK versions have the `wb` device?

**Comment by eivindj-nordic** [comment]
@jayteemo 0.7.0 versions have the `gd25lb256` flash device while newer devices as 0.9.0 (default version in zephyr) have the `gd25wb256`.

**Pull request merged**